packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Serialization/ScmSerializationInterfaces.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+
+namespace Microsoft.Generator.CSharp.ClientModel
+{
+    internal class ScmSerializationInterfaces : SerializationInterfaces
+    {
+        private readonly CSharpType _modelType;
+        private readonly bool _isStruct;
+
+        public ScmSerializationInterfaces(TypeProvider typeProvider, bool hasJson, bool hasXml) : base(typeProvider, hasJson, hasXml)
```

Comment:
I think we should put ModelTypeProvider here?
Or introduce another abstraction such as `SerializableObject`?
Please note that in future's MPG, resources (which is essentially a client) will have serialization as well.

Reply:
Or I think as Michael mentioned here https://github.com/microsoft/typespec/pull/3313/files#r1598777492
serializations do not have to be public therefore it does not have to be part of our framework.

Reply:
Resolving based on our internal discussion and refactoring. Please see the latest changes.